src/org/zaproxy/zap/extension/ascanrulesAlpha/IntegerOverflow.java

```diff
@@ -0,0 +1,229 @@
+	private static final String MESSAGE_PREFIX = "ascanalpha.integeroverflow.";
+	private static final int PLUGIN_ID = 30003;
+	private static final Logger log = Logger.getLogger(IntegerOverflow.class);
+	
+	@Override
+	public int getId() {
+		return PLUGIN_ID;
+	}
+
+	@Override
+	public String getName() {
+		return Constant.messages.getString(MESSAGE_PREFIX + "name");
+	}
+
+	@Override
+	public String[] getDependency() {
+		return null;
+	}
+
+	@Override
+	public String getDescription() {
+		return Constant.messages.getString(MESSAGE_PREFIX + "desc");
+	}
+
+	@Override
+	public int getCategory() {
+		return Category.INJECTION;
+	}
+
+	@Override
+	public String getSolution() {
+		return Constant.messages.getString(MESSAGE_PREFIX + "soln");
+	}
+
+	@Override
+	public String getReference() {
+		return Constant.messages.getString(MESSAGE_PREFIX + "refs");
+	}
+	
+	public String getOther() {
+		return Constant.messages.getString(MESSAGE_PREFIX + "other");
+	}
+	
+	private String getError(char c) {
+		return Constant.messages.getString(MESSAGE_PREFIX + "error"+c);
+	}
+	
+	@Override
+	public void init() {
+
+	}
+
+	/*
+	 * This method is called by the active scanner for each GET and POST parameter for every page 
+	 * @see org.parosproxy.paros.core.scanner.AbstractAppParamPlugin#scan(org.parosproxy.paros.network.HttpMessage, java.lang.String, java.lang.String)
+	 */
+	@Override
+	public void scan(HttpMessage msg, String param, String value) {
+	
+		if (checkStop() == true) return;
+		try {
+			
+			msg = getNewMsg();
+			String returnAttack = randomIntegerString(4);// The number of full length ints to send.
+			setParameter(msg, param, returnAttack);
+			sendAndReceive(msg);			
+			if (attackVector(msg, param, '1', returnAttack) == true) return;
+    		msg = getNewMsg();
+			returnAttack = singleString(4,'0');// The number of full length ints to send.
+			setParameter(msg, param, returnAttack);
+			sendAndReceive(msg);
+			if (attackVector(msg, param, '2', returnAttack) == true) return;
+    		msg = getNewMsg();
+			returnAttack = singleString(4,'1');// The number of full length ints to send.
+			setParameter(msg, param, returnAttack);
+			sendAndReceive(msg);
+			if (attackVector(msg, param, '3', returnAttack) == true) return;
+    		msg = getNewMsg();
+			returnAttack = singleString(4,'9');// The number of full length ints to send.
+			setParameter(msg, param, returnAttack);
+			sendAndReceive(msg);
+			if (attackVector(msg, param, '4', returnAttack) == true) return;
+				return;	
+
+			
+		} catch (Exception e) {
+			log.debug(e.getMessage(), e);
+		}	
+	}
+
+	@Override
+	public int getRisk() {
+		return Alert.RISK_MEDIUM;
+	}
+
+	@Override
+	public int getCweId() {
+		// The CWE id
+		return 190;
+	}
+
+	@Override
+	public int getWascId() {
+		// The WASC ID
+		return 3;
+	}
+	
+	private String randomIntegerString(int length)
+	{
+		
+		int numbercounter = 0;
+		int character = 0;
+		long charactercounter = 0;
+		int maxcharacter = 11 * length;

```

Comment:
Doing calculation here changes the behaviour of the previous code (built string would be a lot longer).
The calculation should be done using an auxiliary variable or directly as parameter of `StringBuilder` constructor.